pkg/state/fee_validation_test.go

```diff
@@ -52,6 +52,84 @@ func TestAssetScriptExtraFee(t *testing.T) {
 	assert.NoError(t, err, "checkMinFeeWaves() failed with valid Burn fee")
 }
 
+/* Negative test
+The account script is set on blockID2, then rollback returns storage to the blockID1.
+The account must not have a verifier anymore. However, the filter is false, so invalid data (verifier) will be returned\
+*/
+func TestAccountHasVerifierAfterRollbackFilterFalse(t *testing.T) {
+	to, path := createCheckerTestObjects(t)
+	defer func() {
+		to.stor.close(t)
+		err := common.CleanTemporaryDirs(path)
+		assert.NoError(t, err, "failed to clean test data dirs")
+	}()
+
+	tx := createSetScriptWithProofs(t)
+	info := defaultCheckerInfo()
+
+	to.stor.activateFeature(t, int16(settings.SmartAccounts))
+
+	to.stor.addBlock(t, blockID1)
+	to.stor.addBlock(t, blockID2)
+	_, err := to.tc.checkSetScriptWithProofs(tx, info)
+	assert.NoError(t, err, "checkSetScriptWithProofs failed with valid SetScriptWithProofs tx")
+
+	address, err := proto.NewAddressFromPublicKey(to.tc.settings.AddressSchemeCharacter, tx.SenderPK)
+	assert.NoError(t, err, "failed to receive an address from public key")
+
+	txPerformerInfo := &performerInfo{blockID: blockID2}
+	err = to.tp.performSetScriptWithProofs(tx, txPerformerInfo)
+	assert.NoError(t, err, "performSetScriptWithProofs failed with valid SetScriptWithProofs tx")
+
+	hasVerifier, err := to.tp.stor.scriptsStorage.newestAccountHasVerifier(address, false)
```

Comment:
Please, move `filter` argument to the local constant for test to improve readability